Utils/HandlerUtil.py

```diff
@@ -61,26 +61,62 @@
 import base64
 import json
 import time
+
+from xml.etree import ElementTree
 from os.path import join
 from Utils.WAAgentUtil import waagent
 from waagent import LoggerInit
 
 DateTimeFormat = "%Y-%m-%dT%H:%M:%SZ"
 
+MANIFEST_XML = "manifest.xml"
+
 class HandlerContext:
     def __init__(self,name):
         self._name = name
         self._version = '0.0'
         return
 
 class HandlerUtility:
-    def __init__(self, log, error, short_name):
+    def __init__(self, log, error, s_name=None, l_name=None, version=None):
         self._log = log
         self._error = error
-        self._short_name = short_name
 
+        if s_name is None or l_name is None or version is None:
+            (long_name, short_name, ext_version) = self._get_extension_info()
+            self._short_name = short_name
+            self._extension_version = ext_version
+            self._log_prefix = '[%s-%s] ' % (long_name, ext_version)
```

Comment:
That's the whole point of my intention to try to retain the existing behavior as much as possible. I simply don't know how all other extensions were using (or not using) the existing log format, and I just wanted to make sure that this fix becomes available only if each extension owner explicitly specifies the desired extension name & version.